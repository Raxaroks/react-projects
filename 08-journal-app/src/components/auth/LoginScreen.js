import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import validator from 'validator';
import { startGoogleLogin, startLoginWithEmailPassword } from '../../actions/auth';
import { removeError, setError } from '../../actions/ui';
import { useForm } from '../../hooks/useForm';


const LoginScreen = () => {

    const dispatch = useDispatch();
    const { loading } = useSelector( state => state.ui );

    const [ formValues, handleInputChange ] = useForm({
        email: 'devops_user@test.com',
        password: 'password123'
    });

    const { email, password } = formValues;

    const handleLogin = (evt) => {
        evt.preventDefault();

        if ( isFormValid() ) {
            // console.log('El formulario es correcto.');
            dispatch( startLoginWithEmailPassword( email, password ) );    
        }
    }

    const handleGoogleLogin = () => {
        dispatch( startGoogleLogin() );
    }

    const isFormValid = () => {
        if ( !validator.isEmail( email ) ) {
            // console.log('El email no es v치lido');
            dispatch( setError('El email no es v치lido') );
            return false;
        }

        else if ( password < 5 ) {
            // console.log('La contrase침a no es segura o no coincide.');
            dispatch( setError('La contrase침a no es segura o no coincide.') );
            return false;
        }

        dispatch( removeError() );
        return true;
    }

    return (
        <>
            <h3 className="auth__title">Login</h3>
            <form onSubmit={ handleLogin }>
                <input
                    type="text"
                    placeholder="Email"
                    name="email"
                    className="auth__input"
                    autoComplete="off"
                    value={ email }
                    onChange={ handleInputChange }
                />
                <input
                    type="password"
                    placeholder="Password"
                    name="password"
                    className="auth__input"
                    value={ password }
                    onChange={ handleInputChange }
                />
                <button
                    className="btn btn-primary btn-block"
                    type="submit"
                    disabled={ loading }
                >
                    Login
                </button> 

                <div className="auth__social-networks">
                    <p><strong>Login con redes sociales</strong></p>
                    <div className="google-btn" onClick={ handleGoogleLogin }>
                        <div className="google-icon-wrapper">
                            <img 
                                className="google-icon"
                                src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
                                alt="Logo de Google"
                            />
                        </div>
                        <p className="btn-text">
                            <b>Login con Google</b>
                        </p>
                    </div>
                </div>

                <Link className="link" to="/auth/register">
                    Crea una nueva cuenta
                </Link>
            </form>
        </>
    );
}

export default LoginScreen;
