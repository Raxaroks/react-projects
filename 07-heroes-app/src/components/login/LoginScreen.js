import React, { useContext } from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

const LoginScreen = ({ history }) => {

    const { dispatch } = useContext(AuthContext); 

    const handleLogin = () => {

        const lastPath = localStorage.getItem('lastPath') || '/';

        dispatch({
            type: types.login,
            payload: {
                name: 'Raxaroks420'
            }
        });
        history.replace( lastPath );

    }

    return (
        <div className="container mt-5">
            <h1>Login</h1> <hr/>

            <button
                style={ { borderRadius: 18 } }
                className="btn btn-primary"
                onClick={ handleLogin }
            >
                Iniciar sesi√≥n
            </button>
        </div>
    );
}

export default LoginScreen;
