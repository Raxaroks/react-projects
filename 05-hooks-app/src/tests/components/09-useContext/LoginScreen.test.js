import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import LoginScreen from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Pruebas en el componente: LoginScreen', () => {

    const setUser = jest.fn();

    const componentWrapper = mount(
        <UserContext.Provider value={{ setUser }}>
            <LoginScreen/>
        </UserContext.Provider>
    );

    test('Debe de renderizarse correctamente.', () => {
        expect(componentWrapper).toMatchSnapshot();
    });    

    test('Debe de ejecutar la funciÃ³n setUser con el argumento esperado.', () => {
        componentWrapper.find('button').prop('onClick')();

        expect(setUser).toHaveBeenCalledWith({
            id: 123,
            name: 'Raxaroks'
        });
    });
    
});
