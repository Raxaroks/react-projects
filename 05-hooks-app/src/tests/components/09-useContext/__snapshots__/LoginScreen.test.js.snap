// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente: LoginScreen Debe de renderizarse correctamente. 1`] = `
<div>
  <h1>
    LoginScreen: PÃ¡gina de login
  </h1>
   
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
